### ISSUE TITLE
Invoke-CMApplyDriverPackage.ps1 3.0.4 - Incorrect Target OS Build Version for 1909

### ISSUE THREAD

**Issue by DreamreaperDX** [MSEndpointMgr/ConfigMgr#226]
Matching during the ConfigMgr WebService phase fails to properly detect Windows 10 1909 mistaking it for 1903 thus causing it to not match any driver packages.  I am deploying 10.0.18363.778 based off the current 1909.3 .ISO in VLSC (serviced by WIM Witch with) but for some reason it thinks it is 10.0.18362.778.  

See below for an example:

![image](https://user-images.githubusercontent.com/9938212/80040388-96180d00-84bf-11ea-9057-dbc95c2a3f84.png)

Winver:
![image](https://user-images.githubusercontent.com/9938212/80041246-8bf70e00-84c1-11ea-8733-feadd0a42bee.png)

However, even in SCCM the .WIM says 18362.778 despite being an 18363 build from VLSC.
![image](https://user-images.githubusercontent.com/9938212/80041476-2e16f600-84c2-11ea-9c44-5159cd8b1859.png)
![image](https://user-images.githubusercontent.com/9938212/80041486-32dbaa00-84c2-11ea-9bff-a7e57abe5657.png)

Additionally, I found some resources that may explain this behavior -- sort of?
Under [Versioning](https://techcommunity.microsoft.com/t5/windows-it-pro-blog/windows-10-version-1909-delivery-options/ba-p/1002660), "Because of the common baseline, even after installing Windows 10, version 1909, individual system file versions will still appear as 18362."

I'm wondering if this matching method is problematic for 1903/1909 based builds and needs to be modified to query some other data set within the operating system.  Has anyone else run into this on 1903/1909?

**Comment by dsolodow** [comment]
I've run into this as well. It's because of the version property on the WIM, so probably would have to find some other way to detect it. I know WIMWitch had similar issues.

**Comment by NickolajA** [comment]
We introduced a fix for this a couple of versions back, so simply add the following to your 1909 TS step parameters for Modern Driver Management:

-TargetOSVersion "18363"

This will override the version returned from the web service and force it to match with 1909 that's kept in a static table within the script.

**Comment by azulol** [comment]
Hey i got the same problem.
with the parameter -TargetOSVersion "18363" nothing changed.
In which order do i have to put this in line?
i got this: -URI "http://www.blablar.asmx" -SecretKey "12345" -Filter "Drivers" -TargetOSVersion "18363"

**Comment by dsolodow** [comment]
I don't have the quotes around the number of -TargetOSVersion but since it's a string I wouldn't think it would matter.

One thing I did find is that if you mount & then dismount with commit the WIM on a machine running a 19600+ build of Windows 10, it corrects the OSVersion property of the WIM

**Comment by NickolajA** [comment]
Yeah this issue is not there if you service a WIM of version 1909 with ADK 2004 that's about to be released soon.

**Comment by NickolajA** [comment]
> Hey i got the same problem.
> 
> with the parameter -TargetOSVersion "18363" nothing changed.
> 
> In which order do i have to put this in line?
> 
> i got this: -URI "http://www.blablar.asmx" -SecretKey "12345" -Filter "Drivers" -TargetOSVersion "18363"

Did you update to use he latest version of the script? It's been running well here. What's in your ApplyDriverPackage.log?

**Comment by azulol** [comment]
Yes i have the latest script version 3.0.4 and thats in the log.
 - Target operating system name detected as: Windows 10	ApplyDriverPackage	08.05.2020 02:09:12	2016 (0x07E0)
 - Target operating system architecture detected as: x64	ApplyDriverPackage	08.05.2020 02:09:12	2016 (0x07E0)
 - Target operating system build version detected as: 10.0.18362.778	ApplyDriverPackage	08.05.2020 02:09:12	2016 (0x07E0)
 - Target operating system translated version detected as: 1903	ApplyDriverPackage	08.05.2020 02:09:12	2016 (0x07E0)

**Comment by phillipwp1** [comment]
Hi. I noticed the same issue at a customer today that recently started buying HP laptops. They normally use only Dells. I added the TargetOSVersion switch. Will let you know how it goes.
That switch should not affect Dell right? As dell only has 3 detection checks compared to HP which has 4?
